### PR TITLE
Parallelise calculation

### PR DESCRIPTION
**Pull request by JamesCaldon**
Will need to format the output array using reshape probably

### CODE REVIEW

**Comment by JamesCaldon** [comment]
Before we merge, we should find a way of changing the format of the output to match the others

File: deformation_measurement/DIC_NR_images.py [hunk 1]
```diff
@@ -200,24 +259,37 @@ def calculate(self):
...
+			raise error("Tried to run calculate before setting parameters. Please use set_parameters first.")
+
+		self.DEFORMATION_PARAMETERS = np.zeros((self.Y_size,self.X_size,12), dtype = float)
+
+		calc_start_time = datetime.now()
+		if (self.parallel == False):
+			self.sequential_calculate(calc_start_time)
+		else:
+			num_cores = multiprocessing.cpu_count()
+			self.DEFORMATION_PARAMETERS = Parallel(n_jobs=num_cores)(delayed(self.parallel_calculate_helper)(i, j, calc_start_time) for i in range(self.Xmin, self.Xmax) for j in range(self.Ymin,self.Ymax))
```
Comment (texm) [inline deformation_measurement/DIC_NR_images.py]: I'm not super keen on having this all on one line

Comment (JamesCaldon) [reply deformation_measurement/DIC_NR_images.py]: Its unfortunately the only way...

Comment (JamesCaldon) [reply deformation_measurement/DIC_NR_images.py]: List comprehension is the only way it parallelizes from my limited research

Comment (texm) [reply deformation_measurement/DIC_NR_images.py]: Ah well as long as it works haha

**Pull request merged**